src/MOI/MOI_wrapper.jl

```diff
@@ -2165,22 +2165,28 @@ function MOI.optimize!(model::Optimizer)
     model.cached_solution.solve_time = time() - start_time
 
     if Xpress.is_mixedinteger(model.inner)
-        #Xpress.getmipsol
-        Xpress.Lib.XPRSgetmipsol(model.inner,
+        Xpress.Lib.XPRSgetmipsol(
+            model.inner,
             model.cached_solution.variable_primal,
-            model.cached_solution.linear_primal) # TODO
-            fill!(model.cached_solution.linear_dual, NaN)
-            fill!(model.cached_solution.variable_dual, NaN)
+            model.cached_solution.linear_primal,
+        )
+        fill!(model.cached_solution.linear_dual, NaN)
+        fill!(model.cached_solution.variable_dual, NaN)
     else
-        Xpress.Lib.XPRSgetlpsol(model.inner,
+        Xpress.Lib.XPRSgetlpsol(
+            model.inner,
             model.cached_solution.variable_primal,
-            model.cached_solution.linear_primal, # TODO
+            model.cached_solution.linear_primal,
             model.cached_solution.linear_dual,
-            model.cached_solution.variable_dual)
+            model.cached_solution.variable_dual,
+        )
     end
+    # TODO(odow): get an Xpress license to test issue https://github.com/jump-dev/Xpress.jl/issues/91 properly.
     rhs = Xpress.getrhs(model.inner)
-
-    model.cached_solution.linear_primal .= rhs .- model.cached_solution.linear_primal
+    for i = 1:min(length(rhs), length(model.cached_solution.linear_primal))
+        x = model.cached_solution.linear_primal[i]
+        model.cached_solution.linear_primal[i] = rhs[i] - x
+    end
```

Comment:
this is not 100%, when the original version fails we might be getting bad results.
I will open an issue.
I have a candidate fix.